src/lib/chips/chip-list.ts

```diff
@@ -734,4 +738,19 @@ export class MatChipList extends _MatChipListMixinBase implements MatFormFieldCo
       }
     });
   }
+
+  /** Checks whether an event comes from inside a chip element. */
+  private _originatesFromChip(event: Event): boolean {
+    let currentElement = event.target as HTMLElement | null;
+
+    while (currentElement && currentElement !== this._elementRef.nativeElement) {
+      if (currentElement.classList.contains('mat-chip')) {
```

Comment:
shouldn't we also check if it originates from the chip input?

Reply:
That's not necessary, because the `focus` method checks whether the input is focused.